src/Microsoft.ComponentDetection.Orchestrator/Services/DetectorProcessingService.cs

```diff
@@ -67,6 +69,21 @@ public class DetectorProcessingService : IDetectorProcessingService
                 var componentRecorder = new ComponentRecorder(this.logger, !detector.NeedsAutomaticRootDependencyCalculation);
                 var isExperimentalDetector = detector is IExperimentalDetector && !(detectorRestrictions.ExplicitlyEnabledDetectorIds?.Contains(detector.Id)).GetValueOrDefault();
 
+                var cancellationToken = CancellationToken.None;
+                if (settings.Timeout != null && settings.Timeout > 0)
+                {
+                    var cts = new CancellationTokenSource();
+                    cancellationToken = cts.Token;
+
+                    // max timeout is equal to settings timeout for non-experimental, and min of settings timeout or 4 minutes for experimenatal
+                    var timeoutSeconds = isExperimentalDetector
```

Comment:
If we're doing this at the top level, do we want to introduce a small buffer (5 seconds less than timeout?) for the processes to exit so we don't hit a task timeout?

Reply:
yep great point